superset/dashboards/dao.py

```diff
@@ -200,7 +200,7 @@ def set_dash_metadata(  # pylint: disable=too-many-locals
             slice_ids = [
                 value.get("meta", {}).get("chartId")
                 for value in positions.values()
-                if isinstance(value, dict)
+                if isinstance(value, dict) and value.get("type") == "CHART"
```

Comment:
See above. Without this logic, when editing a dashboard all the old filter-box charts would be re-added given that the markdown elements still contain the `chartId` field.